File: app/src/main/java/com/danielmaia/desafioandroid/view/adapter/PullAdapter.java
package com.danielmaia.desafioandroid.view.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.danielmaia.desafioandroid.App;
import com.danielmaia.desafioandroid.R;
import com.danielmaia.desafioandroid.model.Pull;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class PullAdapter extends RecyclerView.Adapter<PullAdapter.ViewHolder> {

    private PullAdapter.PullAdapterListener listener;
    private List<Pull> pullList;

    public interface PullAdapterListener {
        void onItemClick(String url);
    }

    public PullAdapter(PullAdapter.PullAdapterListener l, List<Pull> r) {
        listener = l;
        pullList = r;
    }

    @Override
    public PullAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new PullAdapter.ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list_pull, parent, false), listener);
    }

    @Override
    public void onViewDetachedFromWindow(@NonNull ViewHolder holder) {
        super.onViewDetachedFromWindow(holder);
        holder.itemView.clearAnimation();
    }

    @Override
    public void onBindViewHolder(PullAdapter.ViewHolder holder, int position) {
        final Pull pull = pullList.get(position);

        holder.txtTitle.setText(pull.getTitle());
        holder.txtDesc.setText(pull.getBody());
        holder.txtDesc.setVisibility(pull.getBody().length() > 0 ? View.VISIBLE : View.GONE);
        holder.txtUsername.setText(pull.getOwner());

        Glide.with(App.getInstance())
                .load(pull.getAvatar())
                .placeholder(App.getInstance().getResources().getDrawable(R.drawable.ic_profile))
                .into(holder.imgProfile);

        holder.llContainer.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onItemClick(pull.getUrl());
            }
        });
    }

    @Override
    public int getItemCount() {
        return pullList.size();
    }

    public void setList(List<Pull> list) {
        pullList = list;
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        public RelativeLayout llContainer;
        public TextView txtTitle, txtDesc, txtUsername, txtName;
        public CircleImageView imgProfile;
        public PullAdapter.PullAdapterListener listener;

        public ViewHolder(View itemView, final PullAdapter.PullAdapterListener listener) {
            super(itemView);
            this.listener = listener;

            llContainer = itemView.findViewById(R.id.llContainer);
            txtTitle = itemView.findViewById(R.id.txtTitle);
            txtDesc = itemView.findViewById(R.id.txtDesc);
            imgProfile = itemView.findViewById(R.id.imgProfile);
            txtUsername = itemView.findViewById(R.id.txtUsername);
            txtName = itemView.findViewById(R.id.txtName);
        }
    }

}

File: app/src/main/java/br/com/andreyneto/desafioandroid/ui/pulls/PullAdapter.kt
package br.com.andreyneto.desafioandroid.ui.pulls

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import br.com.andreyneto.desafioandroid.R
import br.com.andreyneto.desafioandroid.model.Pull
import br.com.andreyneto.desafioandroid.ui.components.loadImage
import kotlinx.android.synthetic.main.list_pull.view.*


class PullAdapter(private val pulls: List<Pull>,
                  private val ctx: Context,
                  private val presenter: PullsContract.Presenter) : RecyclerView.Adapter<PullAdapter.ViewHolder>() {
    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): PullAdapter.ViewHolder {
        return ViewHolder(LayoutInflater.from(ctx).inflate(R.layout.list_pull, p0, false), presenter)
    }

    override fun getItemCount(): Int {
        return pulls.size
    }

    override fun onBindViewHolder(p0: PullAdapter.ViewHolder, p1: Int) {
        p0.bindView(pulls[p1], pulls.size==p1+1)
    }

    class ViewHolder(itemView: View, val presenter: PullsContract.Presenter): RecyclerView.ViewHolder(itemView) {
        fun bindView(pull: Pull, last: Boolean) {
            itemView.lblPullTitle.text = pull.title
            itemView.lblPullDescription.text = pull.body
            itemView.lblOwner.text = pull.user.name
            itemView.separator.visibility = if(last) View.GONE else View.VISIBLE
            itemView.ownerPic.loadImage(pull.user.profilePic)
            itemView.isClickable = true
            itemView.setOnClickListener { presenter.openPull(pull.url, itemView.context) }
        }
    }
}